### PR TITLE
adding configure option dummyqml

### PR DESCRIPTION
**Pull request by demmm**
sidebar entry can be configured and translated
adding a more elaborate qml example
keeping this in dummyqml for now, another commit will follow with
continuation of dummyqml in a more aptly named module

### CODE REVIEW

File: src/modules/dummyqml/DummyQmlViewStep.h [hunk 1]
```diff
@@ -19,20 +20,27 @@
 #ifndef DUMMYQMLVIEWSTEP_H
 #define DUMMYQMLVIEWSTEP_H
 
+#include <PluginDllMacro.h>
```
Comment (adriaandegroot) [inline src/modules/dummyqml/DummyQmlViewStep.h]: This one with "" as well (it's not a system header).

File: src/modules/dummyqml/DummyQmlViewStep.h [hunk 2]
```diff
@@ -19,20 +20,27 @@
...
+#include "locale/TranslatableConfiguration.h"
```
Comment (adriaandegroot) [inline src/modules/dummyqml/DummyQmlViewStep.h]: Yes, sure. It's shorter than declaring the class (which is all you'd need for that one pointer member).

File: src/modules/dummyqml/DummyQmlViewStep.h [hunk 3]
```diff
@@ -19,20 +20,27 @@
...
+#include "utils/CalamaresUtilsSystem.h"
+#include "utils/Variant.h"
```
Comment (adriaandegroot) [inline src/modules/dummyqml/DummyQmlViewStep.h]: Those two should be in the .cpp

File: src/modules/dummyqml/DummyQmlViewStep.h [hunk 4]
```diff
@@ -19,20 +20,27 @@
...
     DummyQmlViewStep( QObject* parent = nullptr );
     virtual ~DummyQmlViewStep() override;
 
-    virtual QString prettyName() const override;
+    QString prettyName() const override;
+    
+    void setConfigurationMap( const QVariantMap& configurationMap ) override;
 
 private:
+    CalamaresUtils::Locale::TranslatedString* m_notesName;  // As it appears in the sidebar
```
Comment (adriaandegroot) [inline src/modules/dummyqml/DummyQmlViewStep.h]: Needs a destructor to clean it up (even if right now, Calamares module destructors are run .. only sometimes)

File: src/modules/dummyqml/DummyQmlViewStep.cpp [hunk 5]
```diff
@@ -28,8 +31,23 @@ DummyQmlViewStep::~DummyQmlViewStep() {}
...
 }
 
+void
+DummyQmlViewStep::setConfigurationMap( const QVariantMap& configurationMap )
+{ 
+    Calamares::QmlViewStep::setConfigurationMap( configurationMap ); // call parent implementation
+    
+    bool qmlLabel_ok = false;
+    auto qmlLabel = CalamaresUtils::getSubMap( configurationMap, "qmlLabel", qmlLabel_ok );
+    if ( qmlLabel_ok )
```
Comment (adriaandegroot) [inline src/modules/dummyqml/DummyQmlViewStep.cpp]: If it's not ok, the returned map is empty, so it won't contain "notes". you can remove this check (or fold it with the nested if).

**Comment by adriaandegroot** [comment]
Changes requested, but the whole looks really good. This makes it one step closer to having a generic "informative QML page". We should keep watching for useful things to fold "upstream" into the QmlViewStep base class, so that they become useful for other QML-UI consumers as well.

**Comment by demmm** [comment]
Will try and resolve all.  Start a new PR and remove this one?
And for 
>We should keep watching for useful things to fold "upstream" into the QmlViewStep base class

I have opening an outside file (not part of Calamares, can be local or on web) about working, just can't figure a workaround yet to not need to have the file part of qrc.

**Comment by adriaandegroot** [comment]
- Keep this PR and update it
 - Check 750465153f4d6a329b97b2ed9786423b618411d9, that should load externals properly (there's some trickery with schemes, I think)

(Sorry about being terse, at conference far away from home)

File: src/modules/dummyqml/DummyQmlViewStep.h (hunk 3 above)
Comment (demmm) [reply src/modules/dummyqml/DummyQmlViewStep.h]: Fails to build if I move #include "utils/CalamaresUtilsSystem.h" and #include "utils/Variant.h" from the header to cpp file.

File: src/modules/dummyqml/DummyQmlViewStep.h (hunk 4 above)
Comment (demmm) [reply src/modules/dummyqml/DummyQmlViewStep.h]: No idea what to do here

**Comment by demmm** [comment]
Besides my new comments, other issues have been changed, build & tested as working, but since I can't fix all that is needed, I will close this pull request in 1-2 days.

**Comment by adriaandegroot** [comment]
I've merged this into the *qml* branch, needs a little more testing, and would be good to combine with the new descriptor and instancekey work (that's my job, though, so I'm closing the PR as "merged").

**Pull request closed without merging**